src/libraries/System.Private.CoreLib/src/System/Half.cs

```diff
@@ -141,7 +141,7 @@ private ushort Significand
                 // says they should be equal, even if the signs differ.
                 return leftIsNegative || AreZero(left, right);
             }
-            return (short)(left._value) <= (short)(right._value);
+            return left._value <= right._value ^ leftIsNegative;
```

Comment:
Explicit parentheses would be nice, to ensure this is doing the right thing: https://github.com/ucb-bar/berkeley-softfloat-3/blob/master/source/f16_le_quiet.c